IPython/utils/text.py

```diff
@@ -561,7 +561,13 @@ def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
 
                 # eval the contents of the field for the object
                 # to be formatted
-                obj = eval(field_name, kwargs)
+                try:
+                    obj = eval(field_name, kwargs)
+                except NameError:
+                    if supressExceptions:
+                       obj = field_name
```

Comment:
This doesn't work: we need to recover the __original__ string from which field_name was extracted. If '{foo}' was in the `format_string`, `field_name` at this point is 'foo', so we need to put the curly brackets back. Problem is, we don't know where the field_name came from (could have been '$foo' or '{foo}' or.. others?). 

Only sensible way I can see to do this is to change the `Formatter.parse()` interface, make it return the original text as well as `literal_text, field_name, format_spec, conversion` ... something that perhaps should not be taken lightly! How many other things use that method?

I guess we could make a new `parse_with_original()` method (or a less verbose name ;)